android/src/main/java/expo/modules/xmtpreactnativesdk/XMTPModule.kt

```diff
@@ -214,6 +214,17 @@ class XMTPModule : Module() {
             client.canMessage(peerAddress)
         }
 
+        AsyncFunction("staticCanMessage") { peerAddress: String, environment: String, appVersion: String? ->
+            try {
+                logV("staticCanMessage")
+                val options = ClientOptions(api = apiEnvironments(environment, appVersion))
+                val canMessage = Client.canMessage(peerAddress = peerAddress, options = options)
+                canMessage
```

Comment:
There isn't any reason to store this you can just run it like this and that will return correctly.

```suggestion
                Client.canMessage(peerAddress = peerAddress, options = options)
```